components/animation/index.js

```diff
@@ -131,35 +141,33 @@ AFRAME.registerComponent('animation', {
     data.pauseEvents.map(function (eventName) {
       el.removeEventListener(eventName, self.pauseAnimationBound);
     });
+    data.resumeEvents.map(function (eventName) {
+      el.removeEventListener(eventName, self.resumeAnimationBound);
+    });
+    data.restartEvents.map(function (eventName) {
+      el.removeEventListener(eventName, self.restartAnimationBound);
+    });
   },
 
   playAnimation: function () {
```

Comment:
Do we want to leave the play/pause/resume/restart methods in their own function? This will enable the code to be more expandable for future edits if we want to add additional logic in each method